raiden/message_handler.py

```diff
@@ -338,6 +339,13 @@ def handle_message_lockedtransfer(
         sender = from_transfer.balance_proof.sender
 
         if message.target == TargetAddress(raiden.address):
+            encrypted_secret = getattr(message.metadata, "encrypt_secret", None)
+            if encrypted_secret:
+                try:
+                    secret = decrypt_secret(encrypted_secret, raiden.rpc_client.privkey)
+                    return [ReceiveSecretReveal(secret=secret, sender=message.sender)]
+                except InvalidSecret:
+                    log.error("Ignoring invalid encrypted secret")
```

Comment:
It would be good to also log the sender address. 

The question is we should drop the transfer in this case since there's obviously something weird going on.

Reply:
It might be worth, at least on a first run, to fall back to request the secret if the encrypted secret received was invalid. Considering that this failure is logged, I would still think it's worth to keep the fallback behaviour on the first version of the feature.